### ISSUE TITLE
Installing mdns on Windows 7, node-gyp rebuild fails.

### ISSUE THREAD

**Issue by MichelPaus** [agnat/node_mdns#136]
Hello,

I've been trying to get a Meteor install running, but it keeps on breaking while updating mdns dependencies.  
More specifically on `node-gyp rebuild`
I get the following log output:  

    C:\xampp\htdocs\Github\panini>meteor
    [[[[[ C:\xampp\htdocs\Github\panini ]]]]]

    => Started proxy.
    => Started MongoDB.
    soixantecircuits:mdns: updating npm dependencies -- mdns...
    npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
    description
    npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
    repository field.
    npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
    README data
    gypnpm ERR! Windows_NT 6.1.7601
    npm ERR! argv
    "C:\\Users\\Michiel\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.1.3\\mt-o
    s.windows.x86_32\\dev_bundle\\bin\\\\node.exe"
    "C:\\Users\\Michiel\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.1.3\\mt-o
    s.windows.x86_32\\dev_bundle\\bin\\node_modules\\npm\\bin\\npm-cli.js"
    "install"
    npm ERR! node v0.10.36
    npm ERR! npm  v2.7.3
    npm ERR! code ELIFECYCLE
    
    npm ERR! mdns@2.2.2 install: `node-gyp rebuild`
    npm ERR! Exit status 7
    npm ERR!
    npm ERR! Failed at the mdns@2.2.2 install script 'node-gyp rebuild'.
    npm ERR! This is most likely a problem with the mdns package,
    npm ERR! not with npm itself.
    npm ERR! Tell the author that this fails on your system:
    npm ERR!     node-gyp rebuild
    npm ERR! You can get their info via:
    npm ERR!     npm owner ls mdns
    npm ERR! There is likely additional logging output above.
    
    => Errors prevented startup:
    
    While building package soixantecircuits:mdns:
    error: couldn't install npm packages from npm-shrinkwrap
    
    => Your application has errors. Waiting for file change.  

The `soixantecircuits:mdns` package is simply a wrapper around `mdns`, calling that package.  

Thank you for your help

**Comment by y-lohse** [comment]
Do you meet the [requirements for node-gyp](https://github.com/TooTallNate/node-gyp)? The yare a bit extensive on Windows.

**Comment by MichelPaus** [comment]
I'm sorry, I forgot to mention that.
Was struggling for a while already.

I do believe I did most of these things.
 - installed and coupled python 2.7 instead v.3
 - Got Bonjour SDK
 - Installed VS 2010 Express
 - Installed 64bit SDK and compiler update: SDK 7.1

I'm not sure if node-gyp picks up the right version of Visual Studio. My log output doesn't give any errors regarding that, as other people do seem to get. This could mean node-gyp plays nicely with Visual Studio, or not at all..  
I don't know how to test this?

**Comment by y-lohse** [comment]
Yeah that was going to be my second question. I'm also on WIndows 7 and whenever I want to compile node modules with node-gyp, I need to use the console bundled with VisualStudio instead of the regular one, as it has access to extra variables. 
It's located in `C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat` or whatever the path to your VS installation is. Maybe you can try installing mdns using that console, see if that works better.

**Comment by MichelPaus** [comment]
I uninstalled VS2010 and then installed Visual Studio 2013 Express.  
Using `vcvarsall.bat`, like you recommended, `mdns` and `node-gyp` installed flawlessly :)  
Thanks alot!

Still struggling do get my Meteor install up and running though..

**Comment by happyiphone** [comment]
Thank so much for this, I've been struggling with this problem for a few hours. By using the VS shell it worked just fine.

**Comment by davidmurdoch** [comment]
Hopefully this will be a time saver for someone else (even though this *is* already mentioned in the install section)...

Install was failing for me on Windows 10 because the `BONJOUR_SDK_HOME` environment variable wasn't available in the `vcvarsall.bat` command prompt (despite it being in listed in Windows' Environment Variables -> System variables). I had to manually set it in the `vcvarsall.bat` cmd prompt by running `$env:BONJOUR_SDK_HOME = "C:\Program Files\Bonjour SDK\"` before running `npm install mdns`.

System Info:

 * Bonjour SDK 3.0
 * Windows 10
 * Python v2.7.10
 * node-gyp v3.3.1
 * Visual Studio 2015 Update 1 ([Visual C++ packages installed](https://msdn.microsoft.com/en-us/library/60k1461a.aspx))
 * npm 3.5.3
 * node 5.4.1

**Comment by akshay20jha** [comment]
> Hello,
> 
> I've been trying to get a Meteor install running, but it keeps on breaking while updating mdns dependencies.
> More specifically on `node-gyp rebuild`
> I get the following log output:
> 
> ```
> C:\xampp\htdocs\Github\panini>meteor
> [[[[[ C:\xampp\htdocs\Github\panini ]]]]]
> 
> => Started proxy.
> => Started MongoDB.
> soixantecircuits:mdns: updating npm dependencies -- mdns...
> npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
> description
> npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
> repository field.
> npm WARN package.json packages-for-meteor-smartpackage-15k5drd@0.0.0 No
> README data
> gypnpm ERR! Windows_NT 6.1.7601
> npm ERR! argv
> "C:\\Users\\Michiel\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.1.3\\mt-o
> s.windows.x86_32\\dev_bundle\\bin\\\\node.exe"
> "C:\\Users\\Michiel\\AppData\\Local\\.meteor\\packages\\meteor-tool\\1.1.3\\mt-o
> s.windows.x86_32\\dev_bundle\\bin\\node_modules\\npm\\bin\\npm-cli.js"
> "install"
> npm ERR! node v0.10.36
> npm ERR! npm  v2.7.3
> npm ERR! code ELIFECYCLE
> 
> npm ERR! mdns@2.2.2 install: `node-gyp rebuild`
> npm ERR! Exit status 7
> npm ERR!
> npm ERR! Failed at the mdns@2.2.2 install script 'node-gyp rebuild'.
> npm ERR! This is most likely a problem with the mdns package,
> npm ERR! not with npm itself.
> npm ERR! Tell the author that this fails on your system:
> npm ERR!     node-gyp rebuild
> npm ERR! You can get their info via:
> npm ERR!     npm owner ls mdns
> npm ERR! There is likely additional logging output above.
> 
> => Errors prevented startup:
> 
> While building package soixantecircuits:mdns:
> error: couldn't install npm packages from npm-shrinkwrap
> 
> => Your application has errors. Waiting for file change.  
> ```
> 
> The `soixantecircuits:mdns` package is simply a wrapper around `mdns`, calling that package.
> 
> Thank you for your help

Dear all you need is just to install Bonjour  from Apple`s website and also set PATH in User variables
